packages/expo-secure-store/android/src/main/java/expo/modules/securestore/AuthenticationHelper.kt

```diff
@@ -0,0 +1,215 @@
+  }
+
+  private fun openAuthenticationPrompt(
+    promise: Promise,
+    options: ReadableArguments,
+    encryptionCallback: EncryptionCallback,
+    cipher: Cipher,
+    gcmParameterSpec: GCMParameterSpec,
+    postEncryptionCallback: PostEncryptionCallback?
+  ) {
+    if (Build.VERSION.SDK_INT < Build.VERSION_CODES.M) {
+      promise.reject(
+        "E_SECURESTORE_AUTH_NOT_AVAILABLE_ERROR",
+        "Biometric authentication requires Android API 23"
+      )
+      return
+    }
+    if (isAuthenticating) {
+      promise.reject(
+        "E_SECURESTORE_AUTH_ERROR",
+        "Authentication is already in progress"
+      )
+      return
+    }
+
+    val biometricManager = BiometricManager.from(context)
+    when (biometricManager.canAuthenticate(BiometricManager.Authenticators.BIOMETRIC_STRONG)) {
+      BiometricManager.BIOMETRIC_ERROR_HW_UNAVAILABLE, BiometricManager.BIOMETRIC_ERROR_NO_HARDWARE -> {
+        promise.reject(
+          "E_SECURESTORE_AUTH_NOT_AVAILABLE_ERROR",
+          "No hardware available for biometric authentication. Use expo-local-authentication to check if the device supports it."
+        )
+        return
+      }
+      BiometricManager.BIOMETRIC_ERROR_NONE_ENROLLED -> {
+        promise.reject(
+          "E_SECURESTORE_AUTH_ERROR",
+          "No biometrics are currently enrolled"
+        )
+        return
+      }
+    }
+
+    val title = options.getString(AUTHENTICATION_PROMPT_PROPERTY, " ")
+
+    val promptInfo = PromptInfo.Builder()
+      .setTitle(title)
+      .setNegativeButtonText(context.getString(android.R.string.cancel))
+      .build()
+    val fragmentActivity = getCurrentActivity() as FragmentActivity?
+    if (fragmentActivity == null) {
+      promise.reject(
+        "E_NOT_FOREGROUND",
+        "Cannot display biometric prompt when the app is not in the foreground"
+      )
+      return
+    }
+
+    uiManager.runOnUiQueueThread(
+      Runnable {
+        isAuthenticating = true
+
+        BiometricPrompt(
+          fragmentActivity,
+          ContextCompat.getMainExecutor(context),
+          object : BiometricPrompt.AuthenticationCallback() {
+            override fun onAuthenticationSucceeded(result: BiometricPrompt.AuthenticationResult) {
+              super.onAuthenticationSucceeded(result)
+              isAuthenticating = false
+
+              val cipher = result.cryptoObject!!.cipher!!
+              handleEncryptionCallback(
+                promise,
+                encryptionCallback,
+                cipher,
+                gcmParameterSpec,
+                object : PostEncryptionCallback {
+                  @Throws(JSONException::class, GeneralSecurityException::class)
+                  override fun run(promise: Promise, result: Any) {
+                    val obj = result as JSONObject
+                    obj.put(REQUIRE_AUTHENTICATION_PROPERTY, true)
+                    postEncryptionCallback?.run(promise, result)
+                  }
+                }
```

Comment:
The `PostEncryptionCallback` at the beginning before `{` can also be omitted. 
```suggestion
                PostEncryptionCallback { promise, result ->
                  val obj = result as JSONObject
                  obj.put(REQUIRE_AUTHENTICATION_PROPERTY, true)
                  postEncryptionCallback?.run(promise, result)
                }
```

Reply:
Done in https://github.com/expo/expo/pull/14512/commits/531d1a47ddd80d2b5758771176f205a0c590ca9a.